python-package/lightgbm/basic.py

```diff
@@ -2507,13 +2507,13 @@ def _compare_params_for_warning(
         compare_result : bool
           Returns whether two dictionaries with params are equal.
         """
-        for k in other_params:
+        for k, v in other_params.items():
```

Comment:
Fix [PLC0206](https://docs.astral.sh/ruff/rules/dict-index-missing-items/#dict-index-missing-items-plc0206) in new Ruff version.